src/clients/python/experimental_api_v2/library/httpclient.py

```diff
@@ -107,198 +104,447 @@ def close(self):
         """
         self._client_stub.close()
 
-    def get_last_request_id(self):
-        """Get the request ID of the most recent request.
-
-        Returns
-        -------
-        int
-            The request ID, or None if a request has not yet been made
-            or if the last request was not successful.
-
-        """
-        return self._last_request_id
-
-    def is_server_live(self):
+    def is_server_live(self, headers={}):
         """Contact the inference server and get liveness.
 
+        Parameters
+        ----------
+        headers: dict
+            Optional dictionary specifying the headers that must be
+            sent alongwith the request.
+
         Returns
         -------
         bool
             True if server is live, False if server is not live.
 
         Raises
         ------
-        InferenceServerException
+        Exception
             If unable to get liveness.
 
         """
-        response = self._client_stub.get("/v2/health/live")
-        self._last_request_id = raise_if_error(response['NV-Status'])
+        response = self._client_stub.get("v2/health/live", headers=headers)
         return response.status_code == 200
 
-    def is_server_ready(self):
+    def is_server_ready(self, headers={}):
         """Contact the inference server and get readiness.
 
+        Parameters
+        ----------
+        headers: dict
+            Optional dictionary specifying the headers that must be
```

Comment:
"... specifying additional HTTP headers to include in the request."

Reply:
Fixed